### PR TITLE
All tasks are ready for review!

### PR DESCRIPTION
**Pull request by qnislavnachev**

### CODE REVIEW

File: exception.iml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$/src" isTestSource="false" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (EmilGeorgiev) [inline exception.iml]: this file should be in .gitignore

File: src/task1/sumator.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package task1;
+
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+public class sumator {
```
Comment (EmilGeorgiev) [inline src/task1/sumator.java]: in java name of classes should begin with Upper case ```Sumator```

File: src/task1/Sumator.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package task1;
+
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+public class Sumator {
```
Comment (EmilGeorgiev) [inline src/task1/Sumator.java]: I think that input data should be pass  as parameter of method sum() we should have:

```java
public class Sumator {
    public double sum(String a, String b) throw .....Exception {
        
    }
}
```

File: src/task2/readAndCheckValue.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package task2;
+
+import java.io.IOException;
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+public class readAndCheckValue {
```
Comment (EmilGeorgiev) [inline src/task2/readAndCheckValue.java]: 1. Name of classes should start with Upper case.
2. Name should not contains verbs (read, check). Verbs are for methods.
3. Words like (and, is, or) should be avoided

Comment (EmilGeorgiev) [inline src/task2/readAndCheckValue.java]: 1. Task have next condition: "Да се създаде нов тип Exception, който да се генерира ако прочетеното число е извън интервала." I can not see your exception.
2. Да се реализира четене от конзолата на числа в различни интервали ([0,100], [10,20],...). You have only one interval [0-100]

File: src/task3and4/myList.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
+package task3and4;
+import java.lang.Exception;
+import java.util.ArrayList;
+import java.util.List;
+
+public class myList {
```
Comment (EmilGeorgiev) [inline src/task3and4/myList.java]: .

File: src/task3and4/myList.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    List newList;
+    int s=5;
```
Comment (EmilGeorgiev) [inline src/task3and4/myList.java]: it is good if size of ```List``` to be passed as parameter of constructor.

File: src/task3and4/myList.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
+package task3and4;
+import java.lang.Exception;
+import java.util.ArrayList;
+import java.util.List;
+
+public class myList {
+    List newList;
```
Comment (EmilGeorgiev) [inline src/task3and4/myList.java]: It is good to use generics https://docs.oracle.com/javase/tutorial/java/generics/types.html. By this way clieant of your class will can to insert and another types of objects not only Integer

File: src/task3and4/myList.java (hunk 5 above)
Comment (EmilGeorgiev) [inline src/task3and4/myList.java]: Да се направят два типа изключения (към задача 2.3):
 - при опит за добавяне на елемент при запълнен вече списък; 
 - при опит за премахване на елемент от празен списък.

I can not see this two exceptions.

File: src/task1/MyException.java [hunk 8]
```diff
@@ -0,0 +1,7 @@
+package task1;
+
+public class MyException extends Exception {
```
Comment (EmilGeorgiev) [inline src/task1/MyException.java]: 1. Rename to InputDataException
2. extends ReuntimeException
3. Pass as parameter ```String message``` in constructor of this class

File: src/task1/Sumator.java [hunk 9]
```diff
@@ -0,0 +1,14 @@
+package task1;
+
+public class Sumator {
+    public double sum(String a, String b) throws MyException {
+        double sum = 0;
+        try {
+            sum = Double.parseDouble(a) + Double.parseDouble(b);
+        } catch (NumberFormatException e) {
+            throw new MyException();
```
Comment (EmilGeorgiev) [inline src/task1/Sumator.java]: throw new InputDataException("Input data not contains only digits")

File: src/task1/Demo.java [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.util.Scanner;
+
+public class Demo {
+    public static void main(String[] args) throws MyException {
+        Sumator math = new Sumator();
+        Scanner scan = new Scanner(System.in);
+
+        System.out.println("Enter values : ");
+        System.out.println(math.sum(scan.nextLine(), scan.nextLine()));
```
Comment (EmilGeorgiev) [inline src/task1/Demo.java]: for {
    System.out.println("Please enter values : ");
    try {
        Double result = math.sum(scan.nextLine(), scan.nextLine())
    } catch(InputDatException e) {
       System.out.println(e.print...)
       continue;
    }
    
    System.out.println(result);
}

**Comment by qnislavnachev** [comment]
My fixes to Exception tasks are ready for review.

File: src/task1/Demo.java [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+        Scanner scan = new Scanner(System.in);
+
+        System.out.println("Enter values: ");
+        try{
+        result = math.sum(scan.nextLine(), scan.nextLine());
+        }catch (Exception e){
+            throw new WrongValue("This is wrong data");
+        }
+        System.out.println(result);
+    }
```
Comment (EmilGeorgiev) [inline src/task1/Demo.java]: Here client can only one time to pass input data. If data is wring it should start application again.
We should show on user message that input data which it is passed are wrong  and give him a chance to try a second time.

```java
public static void main(String[] args) {
    for {
         System.out.println("Please enter values : ");
         Double result = null;
         try {
             result = math.sum(scan.nextLine(), scan.nextLine())
         } catch(InputDatException e) {
             System.out.println(e.getMessage())
             continue;
        }

        System.out.println(result); 
    }
}
```

File: src/task1/WrongValue.java [hunk 12]
```diff
@@ -0,0 +1,8 @@
+package task1;
+
+public class WrongValue extends RuntimeException {
```
Comment (EmilGeorgiev) [inline src/task1/WrongValue.java]: rename to ```InputDataException```

**Comment by qnislavnachev** [comment]
Here is changes about task 1 from exceptions!

**Comment by EmilGeorgiev** [comment]
LGTM

**Pull request merged**